hudi-common/src/main/java/org/apache/hudi/common/model/PartitionBucketIndexHashingConfig.java

```diff
@@ -0,0 +1,251 @@
+    return loadHashingConfig(storage, hashingConfig.getPath());
+  }
+
+  public static Option<PartitionBucketIndexHashingConfig> loadHashingConfig(HoodieStorage storage, StoragePath hashingConfig) {
+    if (hashingConfig == null) {
+      return Option.empty();
+    }
+    try (InputStream is = storage.open(hashingConfig)) {
+      byte[] content = FileIOUtils.readAsByteArray(is);
+      return Option.of(PartitionBucketIndexHashingConfig.fromBytes(content));
+    } catch (IOException e) {
+      LOG.error("Error when loading hashing config, for path: " + hashingConfig.getName(), e);
+      throw new HoodieIOException("Error while loading hashing config", e);
+    }
+  }
+
+  /**
+   * TODO zhangyue19921010 need more test and details java Doc
+   */
+  public static String getHashingConfigInstantToLoad(HoodieTableMetaClient metaClient) {
+    List<String> instants = metaClient.getActiveTimeline().getCompletedReplaceTimeline()
+        .getInstants().stream().map(HoodieInstant::requestedTime).collect(Collectors.toList());
+    Option<HoodieInstant> earliestInstant = metaClient.getActiveTimeline().getCommitsTimeline().filterCompletedInstants().firstInstant();
+    String instantToLoad = "";
+    try {
+      List<String> hashingConfigInstants = metaClient.getStorage()
+          .listDirectEntries(new StoragePath(metaClient.getHashingMetadataConfigPath())).stream().map(info -> {
+            String instant = getHashingConfigInstant(info.getPath().getName());
+            if (StringUtils.isNullOrEmpty(instant)) {
+              throw new HoodieException("Failed to get hashing config instant to load.");
+            }
+            return instant;
+          }).sorted().collect(Collectors.toList());
+
+      for (String instant : hashingConfigInstants) {
+        if (!earliestInstant.isPresent()) {
+          instantToLoad = instant;
+          break;
+        } else if (instants.contains(instant)) {
+          instantToLoad = instant;
+          break;
+        } else if (instant.compareTo(earliestInstant.get().requestedTime()) < 0) {
+          instantToLoad = instant;
+          break;
+        }
+      }
+
+      if (StringUtils.isNullOrEmpty(instantToLoad)) {
+        throw new HoodieException("Failed to get hashing config instant to load.");
+      }
+
+      return instantToLoad;
+    } catch (IOException e) {
+      throw new HoodieIOException("Failed to get hashing config instant to load.", e);
+    }
+  }
+
+  public static PartitionBucketIndexHashingConfig loadingLatestHashingConfig(HoodieTableMetaClient metaClient) {
+    String instantToLoad = getHashingConfigInstantToLoad(metaClient);
+    Option<PartitionBucketIndexHashingConfig> latestHashingConfig = loadHashingConfig(metaClient.getStorage(), getHashingConfigPath(metaClient.getBasePath().toString(), instantToLoad));
+    ValidationUtils.checkArgument(latestHashingConfig.isPresent(), "Can not load latest hashing config " + instantToLoad);
+
+    return latestHashingConfig.get();
```

Comment:
return the option should be better to avoid NPEs.

Reply:
here we need to make sure loading hashing config success each call. Also add `ValidationUtils.checkArgument(latestHashingConfig.isPresent())` to guard it.